modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/PlannerHelper.java

```diff
@@ -226,7 +227,7 @@ private IgniteRel processScan(TableScan scan) {
                 return (IgniteRel) scan;
             }
 
-            Set<Integer> indexedCols = Set.copyOf(
+            Set<Integer> indexedCols = CollectionUtils.setOf(
```

Comment:
```suggestion
            IntSet indexedCols = CollectionUtils.setOf(
```
you may change stream to `mapToInt().contains()`